src/odemis/driver/picoquant.py

```diff
@@ -32,52 +37,123 @@
+PH_CHANOFFSMAX = 8000  # ps
+
+PH_ACQTMIN = 1  # ms
+PH_ACQTMAX = 360000000  # ms  (100*60*60*1000ms = 100h)
+
+PH_PHR800LVMIN = -1600  # mV
+PH_PHR800LVMAX = 2400  # mV
+
+PH_HOLDOFFMAX = 210480  # ns
+
 
-HISTCHAN = 65536  # number of histogram channels
-TTREADMAX = 131072  # 128K event records
+# Based on hhdefin.h for HydraHarp 400 DLL
+HH_MAXDEVNUM = 8  # max num of USB devices
 
-MODE_HIST = 0
-MODE_T2 = 2
-MODE_T3 = 3
+HH_MAXINPCHAN = 8  # max num of physical input channels
 
-FEATURE_DLL = 0x0001
-FEATURE_TTTR = 0x0002
-FEATURE_MARKERS = 0x0004
-FEATURE_LOWRES = 0x0008
-FEATURE_TRIGOUT = 0x0010
+HH_BINSTEPSMAX = 26  # get actual number via HH_GetBaseResolution() !
 
-FLAG_FIFOFULL = 0x0003  # T-modes
-FLAG_OVERFLOW = 0x0040  # Histomode
-FLAG_SYSERROR = 0x0100  # Hardware problem
+HH_MAXHISTLEN = 65536  # max number of histogram bins/channels (2 ** 16)
+HH_MAXLENCODE = 6  # max length code histo mode
 
-BINSTEPSMAX = 8
+HH_MAXHISTLEN_CONT = 8192  # max number of histogram bins in continuous mode
+HH_MAXLENCODE_CONT = 3  # max length code in continuous mode
 
-SYNCDIVMIN = 1
-SYNCDIVMAX = 8
+HH_MAXCONTMODEBUFLEN = 262272  # bytes of buffer needed for HH_GetContModeBlock
 
-ZCMIN = 0  # mV
-ZCMAX = 20  # mV
-DISCRMIN = 0  # mV
-DISCRMAX = 800  # mV
+HH_TTREADMAX = 131072  # 128K event records can be read in one chunk
+HH_TTREADMIN = 128  # 128 records = minimum buffer size that must be provided
 
-OFFSETMIN = 0  # ps
-OFFSETMAX = 1000000000  # ps
+HH_MODE_HIST = 0
+HH_MODE_T2 = 2
+HH_MODE_T3 = 3
+HH_MODE_CONT = 8
 
-SYNCOFFSMIN = -99999  # ps
-SYNCOFFSMAX = 99999  # ps
+HH_MEASCTRL_SINGLESHOT_CTC = 0  # default
+HH_MEASCTRL_C1_GATED = 1
+HH_MEASCTRL_C1_START_CTC_STOP = 2
+HH_MEASCTRL_C1_START_C2_STOP = 3
+# continuous mode only
+HH_MEASCTRL_CONT_C1_GATED = 4
+HH_MEASCTRL_CONT_C1_START_CTC_STOP = 5
+HH_MEASCTRL_CONT_CTC_RESTART = 6
 
-CHANOFFSMIN = -8000  # ps
-CHANOFFSMAX = 8000  # ps
+HH_EDGE_RISING = 1
+HH_EDGE_FALLING = 0
 
-ACQTMIN = 1  # ms
-ACQTMAX = 360000000  # ms  (100*60*60*1000ms = 100h)
+HH_FEATURE_DLL = 0x0001
+HH_FEATURE_TTTR = 0x0002
+HH_FEATURE_MARKERS = 0x0004
+HH_FEATURE_LOWRES = 0x0008
+HH_FEATURE_TRIGOUT = 0x0010
 
-PHR800LVMIN = -1600  # mV
-PHR800LVMAX = 2400  # mV
+HH_FLAG_OVERFLOW = 0x0001  # histo mode only
+HH_FLAG_FIFOFULL = 0x0002
+HH_FLAG_SYNC_LOST = 0x0004
+HH_FLAG_REF_LOST = 0x0008
+HH_FLAG_SYSERROR = 0x0010  # hardware error, must contact support
+HH_FLAG_ACTIVE = 0x0020  # measurement is running
 
-HOLDOFFMAX = 210480  # ns
+HH_SYNCDIVMIN = 1
+HH_SYNCDIVMAX = 16
+
+HH_ZCMIN = 0  # mV
+HH_ZCMAX = 40  # mV
+HH_DISCRMIN = 0  # mV
+HH_DISCRMAX = 1000  # mV
+
+HH_CHANOFFSMIN = -99999  # ps
+HH_CHANOFFSMAX = 99999  # ps
+
+HH_OFFSETMIN = 0  # ns
+HH_OFFSETMAX = 500000  # ns
+HH_ACQTMIN = 1  # ms
+HH_ACQTMAX = 360000000  # ms  (100*60*60*1000ms = 100h)
+
+HH_STOPCNTMIN = 1
+HH_STOPCNTMAX = 4294967295  # 32 bit is mem max
+
+HH_HOLDOFFMIN = 0  # ns
+HH_HOLDOFFMAX = 524296  # ns
 
 
 class PHError(Exception):
```

Comment:
What is a PHError, is that when the pH (https://en.wikipedia.org/wiki/PH?oldformat=true) is to low/high?
Maybe an description when/why such an error is raised can be added?